service-bus.api.md

```diff
@@ -0,0 +1,256 @@
+
+```ts
+
+import { AmqpMessage } from '@azure/core-amqp';
+import { DataTransformer } from '@azure/core-amqp';
+import { delay } from '@azure/core-amqp';
+import { Delivery } from 'rhea-promise';
+import Long from 'long';
+import { MessagingError } from '@azure/core-amqp';
+import { OperationOptions } from '@azure/core-auth';
+import { RetryOptions } from '@azure/core-amqp';
+import { TokenCredential } from '@azure/core-amqp';
+import { TokenType } from '@azure/core-amqp';
+import { WebSocketImpl } from 'rhea-promise';
+import { WebSocketOptions } from '@azure/core-amqp';
+
+// @public
+export interface CorrelationFilter {
+    contentType?: string;
+    correlationId?: string;
+    label?: string;
+    messageId?: string;
+    replyTo?: string;
+    replyToSessionId?: string;
+    sessionId?: string;
+    to?: string;
+    userProperties?: any;
+}
+
+// @public
+export interface CreateBatchOptions extends OperationOptions {
+    maxSizeInBytes?: number;
+    retryOptions?: RetryOptions;
+}
+
+export { DataTransformer }
+
+// @public
+export interface DeadLetterOptions {
+    deadLetterErrorDescription: string;
+    deadLetterReason: string;
+}
+
+export { delay }
+
+export { Delivery }
+
+// @public
+export interface GetMessageIteratorOptions extends OperationOptions, WaitTimeOptions {
+}
+
+// @public
+export interface GetSessionReceiverOptions extends OperationOptions {
+    maxSessionAutoRenewLockDurationInSeconds?: number;
+    sessionId?: string;
+}
+
+// @public
+export interface MessageHandlerOptions {
+    autoComplete?: boolean;
+    maxConcurrentCalls?: number;
+    maxMessageAutoRenewLockDurationInSeconds?: number;
+}
+
+// @public
+export interface MessageHandlers<ReceivedMessageT> {
+    processError(err: Error): Promise<void>;
+    processMessage(message: ReceivedMessageT): Promise<void>;
+}
+
+export { MessagingError }
+
+// @public
+export interface ReceiveBatchOptions extends OperationOptions, WaitTimeOptions {
+}
+
+// @public
+export interface ReceivedMessage extends ServiceBusMessage {
+    readonly _amqpMessage: AmqpMessage;
+    readonly deadLetterSource?: string;
+    readonly deliveryCount?: number;
+    readonly enqueuedSequenceNumber?: number;
+    readonly enqueuedTimeUtc?: Date;
+    readonly expiresAtUtc?: Date;
+    lockedUntilUtc?: Date;
+    readonly lockToken?: string;
+    readonly sequenceNumber?: Long;
+}
+
+// @public
+export interface ReceivedMessageWithLock extends ReceivedMessage {
+    abandon(propertiesToModify?: {
+        [key: string]: any;
+    }): Promise<void>;
+    complete(): Promise<void>;
+    deadLetter(options?: DeadLetterOptions & {
+        [key: string]: any;
+    }): Promise<void>;
+    defer(propertiesToModify?: {
+        [key: string]: any;
+    }): Promise<void>;
+    renewLock(): Promise<Date>;
+}
+
+// @public
+export interface Receiver<ReceivedMessageT> {
+    close(): Promise<void>;
+    diagnostics: {
```

Comment:
Side thought: After recent customer interviews, I wonder if the word `diagnostics` may be more understood as diagnostics for the receiver and less as diagnostics for the queue/subscription i.e a user might expect to find methods or properties indicating whether the receiver/connection is open or closed or broken or in the process of reconnecting etc